src/index.d.ts

```diff
@@ -17,4 +17,4 @@ export const createContainer: <State, Update, Props>(
 // deep proxy utils
 
 export const trackMemo: (obj: unknown) => void;
-export const getUntracked: <T>(obj: T) => T;
+export const getUntrackedObject: <T>(obj: T) => T;
```

Comment:
since the use of `getUntrackedObject` is limited to objects only the type definition can be improved to:
```
export const getUntrackedObject: <T extends object>(obj: T) => T;
```
Unfortunately, array is a typeof object so it will not throw an error but at least it will exclude the primitive types.

See http://www.typescriptlang.org/play/?ssl=5&ssc=57&pln=5&pc=1#code/JYOwLgpgTgZghgYwgAgMpjpA8gIwFYQJjIDeAUMpcgNoDWEAngFzIDOYUoA5gLotwgGAGgpU6jFiACuAWxzQ+yAcLIBfMjCkgiwAPYhkXCGACq4KInoATXASIAeACrIIAD0ggrrZLvyEwAHwAFL54TI4AlOGkopRQxlJQBqEA3GpkZEam5pYQNn5EQSTIMLq6LACMyKoRaVlmHLn5dmBB1BU8tRn1OQjWtv5BFV09jX15A4UARKW6UyPGDRbjzYNaVhAwoHkL2WP9Ba3SADbHu0tNk60cUhC1QA

Reply:
> since the use of getUntrackedObject is limited to objects

Uh, no, I would like to accept any value. That would make developing custom `dispatch` easier.
It also accepts arrays.

OK, that's a good sign for adding comments. Will do.

Reply:
Do you suggest to change the function name if it accepts non-object values?

Reply:
> Do you suggest to change the function name if it accepts non-object values?
I think it might be for the better to leave it as `getUntrackedObject` since it's intended to be used for objects - might reduce misusage of it.

Reply:
> > since the use of getUntrackedObject is limited to objects
> 
> Uh, no, I would like to accept any value. That would make developing custom `dispatch` easier.
> It also accepts arrays.
> 
> OK, that's a good sign for adding comments. Will do.

How about 2 functions are exported?
`getUntrackedObject` should do exactly what it does -- attempt to untrack an object.
`getUntracked` untrack any value, not limited to objects. 

Or, even better, we can refactor `getUntrackedObject` to omit the `if(isPlainObj(obj))` check and to assume that given parameter is an object,

And introduce another API function `getUntracked` which is safe to call with any value. It's implementation should do the `isPlainObj(obj)` check and call the `getUntrackedObject`.

This way, `getUntrackedObject` will have less overhead for direct users.

Reply:
Thanks for you suggestion. But, I wouldn't take that approach because of various reasons.

1) I would prefer to export less functions as possible. `getUntrackedObject` is much safer and wouldn't be misused.
2) I would like to hide implementation details. `isPlainObject` is actually an implementation detail. I could imagine that would be changed in the future. So, I don't want users of this library to assume what is an object that would be tracked.
3) As far as I understand, the current implementation of `isPlainObject` is pretty lightweight, and it wouldn't be a bottleneck. If, in the future, we find use cases that this would impact performance, we can consider exporting another function for optimization.

I hope those make sense.

Reply:
> That would make developing custom dispatch easier.

I might be wrong about this. I will try it in an example.

Reply:
Hm, it can be easier for getUntrackedObject to return `null` if `obj` is not a proxy.